x-pack/plugins/observability_solution/entity_manager/server/lib/entities/transform/generate_history_transform.ts

```diff
@@ -21,19 +21,50 @@ import {
   generateHistoryTransformId,
   generateHistoryIngestPipelineId,
   generateHistoryIndexName,
+  generateHistoryBackfillTransformId,
 } from '../helpers/generate_component_id';
+import { isBackfillEnabled } from '../helpers/is_backfill_enabled';
 
 export function generateHistoryTransform(
-  definition: EntityDefinition
+  definition: EntityDefinition,
+  backfill = false
 ): TransformPutTransformRequest {
+  if (backfill && !isBackfillEnabled(definition)) {
+    throw new Error(
+      'This function was called with backfill=true without history.settings.backfillSyncDelay'
+    );
+  }
+
   const filter: QueryDslQueryContainer[] = [];
 
   if (definition.filter) {
     filter.push(getElasticsearchQueryOrThrow(definition.filter));
   }
 
+  if (backfill && definition.history.settings?.backfillLookbackPeriod) {
```

Comment:
these two could potentially conflict if the filter already contains a @timestamp. this is the filters applied with the builtin definition. I can't think of a reliable way to prevent this but we should document favoring the settings instead of the filter when it comes to the lookback period

```
    "query": {
      "bool": {
        "filter": [
          {
            "bool": {
              "should": [
                {
                  "range": {
                    "@timestamp": {
                      "gte": "now-10m"
                    }
                  }
                }
              ],
              "minimum_should_match": 1
            }
          },
          {
            "range": {
              "@timestamp": {
                "gte": "now-1h"
              }
            }
          }
        ]
      }
    }
  }
```

Reply:
We should add `lookbackPeriod` back to the history object, eventually when we expose custom definitions to the user and then use `@timestamp` (or `timestampField`) in their filter, we can either throw and error OR warning letting them know that we are adding `lookbackPeriod` and `backfillLookbackPeiod` and their timestamp filter will likely interfere.